tests/test_core.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from unittest.mock import MagicMock
+
+from auto_smart_open.core import FileStorageBackend
+
+class Mocker:
+    def __enter__(self):
+        return "entering context"
+    def __exit__(self, type, value, traceback):
+        raise FileNotFoundError
+
+def test_sopen():
+    """
+    Replace the smart_open open with a mock function to see if __enter__ and __exit__ methods are properly called.
+    """
+    backend = FileStorageBackend()
+    # mock the smart open open
+    backend.open = MagicMock(return_value=Mocker())
```

Comment:
Use `unittest.mock.patch` function to change behavior of `smart_open.open` function.

Example with default value:
```
with unittest.mock.patch('math.sqrt', return_value=2) as _:
    assert math.sqrt(6767) == 2
```

Example with special function (can be used to set a market for example):
```
mock_called = False
def myMock(x):
    global mock_called
    mock_called = True
    return x**2

with unittest.mock.patch('math.sqrt', new=myMock) as _:
    assert math.sqrt(6767) == 6767**2
    assert mock_called
```

Reply:
https://docs.python.org/3/library/unittest.mock.html#quick-guide I thought MagiMock would do this ?